### ISSUE TITLE
Show repository facts on GitHub Enterprise

### ISSUE THREAD

**Issue by timmjd** [squidfunk/mkdocs-material#5157]
### Context

Displaying the `repository stats` like stars / forks / last release does not work for `OnPrem` installations of `GitHub Enterprise`.

### Bug description

If we follow the documentation about [adding a git repository](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/):

> If your documentation is related to source code, Material for MkDocs provides the ability to display information to the project's repository as part of the static site, including stars and forks. Furthermore, the [date of last update and creation](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#document-dates), as well as [contributors](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#document-contributors) can be shown.

It will only display a link but does not display the metadata / stats of the git repository.

This seems the code does only work for content hosted on `github.com` but does not work for `GitHub Enterprise` that are hosted with a custom DNS on an intranet.

The detection about GitHub / GitLab is based on the domain name like `github.com`:
* https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/javascripts/components/source/facts/_/index.ts#L73

The API endpoint is fixed to the public API:
* https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/javascripts/components/source/facts/github/index.ts#L64

To support OnPrem it would somehow require to __enforce the server type__ and also __specify the API endpoint__.

So the API endpoint for OnPrem would be something like `https://github.local/api/v3/repos/${user}/${repo}`

#### GitHub.com
[API endpoint](https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin?apiVersion=2022-11-28)
> `https://api.github.com`

[Pages location](https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/about-github-pages)
> `http(s)://<organization>.github.io/<repository>`

#### GitHub Enterprise
[API endpoint](https://docs.github.com/en/enterprise-server@3.8/rest/enterprise-admin)
> `http(s)://HOSTNAME/api/v3`

[Pages location](https://docs.github.com/en/enterprise-server@3.8/pages/getting-started-with-github-pages/about-github-pages)
> `http(s)://<hostname>/pages/<username>`
> `http(s)://pages.<hostname>/<username>`

### Reproduction

Host MKDocs via pages on an GitHub OnPrem installation

### Steps to reproduce

* Create your MKDocs configuration
* Make sure you have `repo_url` configured
* Upload your content to become a pages

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. Yes, the stars/forks logic only works for public repositories, and I don't see much need to display stars/forks for private/on-premise repositories. Additionally, I don't have access to GitHub Enterprise. Thus I would consider this out-of-scope. If you wish to add this functionality, we can discuss a concrete proposal in the form of a PR, but as I have no way of testing this, I'm not really comfortable adding this to master, as we might break it without knowing in the future.

**Comment by squidfunk** [comment]
Closing as out-of-scope. As said, if you wish to propose a PR, feel free. I can't guarantee a merge though. It depends how involved the proposed solution turns out to be. If you manage to find a good abstraction, it might happen though.

**Comment by dellagustin-sap** [comment]
Just for the purpose of cross-linking https://github.com/squidfunk/mkdocs-material/discussions/5402#discussion-5114978

**Comment by dellagustin-sap** [comment]
Hello @timmjd , I am also interested in having this feature.
I just did a test that showed an extra complication. I sent a GET request to `https:<hostname>/api/v3` from the console of a page hosted on an instance of GitHub Enterprise at `https://pages.<hostname>` and I got an HTTP Error 401 (Unauthorized) as response. Our instance does not allow anonymous access to either the site, the pages or the API.
I don't know if there is an easy way to overcome that.
